loopout.go

```diff
@@ -706,7 +706,77 @@ func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
 		return nil, fmt.Errorf("register spend ntfn: %v", err)
 	}
 
+	var paymentComplete bool
+	pushPreimage := func(ctx context.Context) error {
+		// If our payment is complete, there is no need to proceed in
+		// attempting to push the preimage.
+		if paymentComplete {
+			return nil
+		}
+
+		// If we have not yet revealed our preimage through a sweep, we
+		// do not push the preimage because we may choose to never sweep
+		// if fees are too high.
+		if s.state != loopdb.StatePreimageRevealed {
+			return nil
+		}
+
+		// Create a child context for a one time lookup of our payment.
+		// We defer cancelling it so that lnd will terminate the stream.
+		trackCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		// Lookup our payment in lnd before we push to the server, this
+		// allows us to exit early if the payment is already settled.
+		statusChan, errChan, err := s.lnd.Router.TrackPayment(
+			trackCtx, s.hash,
+		)
+		if err != nil {
+			return err
+		}
+
+		// Get a status update (or error) from our payment tracking or
+		// exit if our context is cancelled.
+		select {
+		// If our payment succeeded, we have no need to push the
+		// preimage and can exit early. We also set paymentComplete to
+		// true so that we do not try to push the preimage again.
+		case status := <-statusChan:
+			if status.State == lnrpc.Payment_SUCCEEDED {
+				s.log.Infof("Payment succeeded, no need to " +
+					"push preimage")
+
+				paymentComplete = true
+				return nil
+			}
+
+			// If our payment has not succeeded, we fallthrough
```

Comment:
Could it be that the payment is still in flight? Should we wait it out in that case?

Reply:
Since the hodl htlc is locked in, I would expect the payment to be in flight at this stage?